src/Illuminate/Foundation/Console/KeyGenerateCommand.php

```diff
@@ -111,6 +119,40 @@ protected function writeNewEnvironmentFileWith($key)
         return true;
     }
 
+    /**
+     * Move the current APP_KEY to APP_PREVIOUS_KEYS.
+     *
+     * @param  string  $envContent
+     * @param  string  $currentKey
+     * @return string
+     */
+    protected function moveToPreviousKeys($envContent, $currentKey)
+    {
+        preg_match('/^APP_PREVIOUS_KEYS=(.*)$/m', $envContent, $matches);
+
+        $existingPreviousKeys = isset($matches[1]) ? trim($matches[1]) : '';
```

Comment:
You can work with match groups here to avoid magic numeric indices

```suggestion
        preg_match('/^APP_PREVIOUS_KEYS=(?<previousKeys>.*)$/m', $envContent, $matches);

        $existingPreviousKeys = isset($matches[1]) ? trim($matches['previousKeys']) : '';
```